### PR TITLE
[IIGO] Made default sanction score a parameter

### PR DESCRIPTION
**Pull request by Neelesh99**
# Summary

Made default sanction score a parameter so we can change it with Sim. Also renamed IIGOSanctionScore to RuleSanctionPenalty, because it was too similar to IIGOSanctionsScore. (There's an extra s)

### CODE REVIEW

File: internal/server/iigointernal/judiciary_test.go [hunk 1]
```diff
@@ -1977,8 +1977,10 @@ func defaultInitJudiciary() judiciary {
 		sanctionRecord:        map[shared.ClientID]shared.IIGOSanctionsScore{},
 		sanctionThresholds:    map[shared.IIGOSanctionsTier]shared.IIGOSanctionsScore{},
 		ruleViolationSeverity: map[string]shared.IIGOSanctionsScore{},
-		gameConf:              &config.IIGOConfig{},
-		gameState:             &gamestate,
+		gameConf: &config.IIGOConfig{
+			DefaultSanctionScore: 5,
+		},
```
Comment (lhl2617) [inline internal/server/iigointernal/judiciary_test.go:1982]: image

Comment (Neelesh99) [reply internal/server/iigointernal/judiciary_test.go:1982]: How do you have these memes ready to go XD

File: params.go [hunk 2]
```diff
@@ -440,6 +446,7 @@ func parseConfig() (config.Config, error) {
 		InspectBallotActionCost:         shared.Resources(*iigoInspectBallotActionCost),
 		InspectAllocationActionCost:     shared.Resources(*iigoInspectAllocationActionCost),
 		AppointNextPresidentActionCost:  shared.Resources(*iigoAppointNextPresidentActionCost),
+		DefaultSanctionScore:            shared.IIGOSanctionsScore(*iigoDefaultSanctionScore),
```
Comment (lhl2617) [inline params.go:449]: idk why the sequence does not match that of the vars above but i'd strive to match it so if you forgot to put it in you know immediately from the seq

File: params.go [hunk 3]
```diff
@@ -346,6 +346,12 @@ var (
 		"Length of the term for the Judge",
 	)
 
+	iigoDefaultSanctionScore = flag.Uint(
```
Comment (lhl2617) [inline params.go]: this implies that the sanctionscore can be changed mid game right

File: internal/server/iigointernal/judiciary_test.go (hunk 1 above)
Comment (Neelesh99) [reply internal/server/iigointernal/judiciary_test.go:1982]: Yeah it's just for a test so it'll always use that value to test that function.

Comment (lhl2617) [reply internal/server/iigointernal/judiciary_test.go:1982]: at fb, we had macros to easily get memes in. I miss that so much. I just google now.

File: params.go (hunk 3 above)
Comment (Neelesh99) [reply params.go]: Yup so judges have the option to upload their own custom scoring for particular rules that they thing should be scored more/less severely than the default. This value is what we revert to if the client judge doesn't give us a custom value for a particular rule.

**Review by lhl2617 — APPROVED** [review]
preapproving but i hope you can arrange the sequence of the params

File: params.go (hunk 2 above)
Comment (Neelesh99) [reply params.go:449]: Yeah fair enough, updated in latest commit

**Pull request merged**